test/parallel/test-stream-pipeline.js

```diff
@@ -1488,3 +1488,27 @@ const tsp = require('timers/promises');
 
   run().then(common.mustCall());
 }
+
+{
+  const s = new PassThrough({ objectMode: true });
+  pipeline(async function*() {
+    await Promise.resolve();
+    yield 'hello';
+    yield 'world';
+    yield 'world';
+  }, s, async function(source) {
+    let ret = '';
+    let n = 0;
+    for await (const chunk of source) {
+      if (n++ > 1) {
+        return;
+      }
+      ret += chunk;
+    }
+    return ret;
+  }, common.mustCall((err, val) => {
+    assert.strictEqual(err, undefined);
+    assert.strictEqual(val, 'helloworld');
```

Comment:
This test seems to be broken. `val` is undefined, I'm not sure why the thrown exception does not make the process exit. I did not investigate.

Reply:
Yea.. that's weird. Fixed the test.

Reply:
Updated test looks good, but why the process did not exit?

Reply:
I can reproduce the same issue with current Node.js version (v17.2.0) so it is not related to this PR but it is something to investigate and fix.

Reply:
It seems the error thrown in the callback is caught and swallowed here https://github.com/nodejs/node/blob/147d23b2116394c5ed183217723ee3025f8ef99b/lib/internal/streams/pipeline.js#L163-L164.

Reply:
Fixed, PTAL, https://github.com/nodejs/node/pull/40881/commits/f0ec29b81563fd93c5951d046c10f804e48f060f